### PR TITLE
Updates for the 2.0.0 version.

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

**Comment by fulldecent** [comment]
Some contracts are being copy-pasted into here from ethereum-utils. Also the same files will exist in a new repository.

For DRY let's decide which version is the official someplace (I recommend the new repository). Then we should link from these files to the original files.

**Comment by MoMannn** [comment]
The ethereum-utils repo will get archived when this is merged since it is not used anywhere else. 

Since this is a reference implementation I believe this should be a stand alone repo not dependant on anything else. If we reference the other repo where will be a lot of unnecessary stuff(for this implementation) that comes with it. 

@xpepermint what do you think?

**Comment by fulldecent** [comment]
To be clear, I'm talking about, for example SafeMath. If we have our own SafeMath copy-paste here and in the new repo then we should cross reference so that any changes will stay synchronized.

**Comment by MoMannn** [comment]
@fulldecent @xpepermint Can you please review changes, after I get the 👍 from both I will update the readme.

**Comment by MoMannn** [comment]
@fulldecent @xpepermint I think we should use this to update to solidity 0.5.x as well. But that would mean we need to update [truffle to 0.5.x which is still in beta](https://github.com/trufflesuite/truffle/releases/tag/v5.0.0-beta.2).  Thoughts?

**Comment by xpepermint** [comment]
Yup! We should upgrade to the latest Solidity version. Since we use Specron in all our repositories, I'd go with Specron.

**Comment by fulldecent** [comment]
Okay I see now that Solidity 0.5.1 is released for real, it is not prerelease.

Also I got a credit in the release notes! https://github.com/ethereum/solidity/releases/tag/v0.5.1 cool

Good to go with using 0.5.1.

---

I don't have any notes on specron, but this issue can get my endorsement https://github.com/specron/framework/issues/46

**Comment by MoMannn** [comment]
@xpepermint @fulldecent Any thoughts about reverts messages? It is not a breaking change so it can get added later but since I'm refactoring anyway I can add them. The question is should I? And if in what form.

**Comment by MoMannn** [comment]
@fulldecent @xpepermint Can you please do another review. I switched to specron, solidity 0.5.1 and made some smaller changes.

**Comment by fulldecent** [comment]
Please explain the intended release strategy.

Proposed:

- Note in README that pre-release software is not eligible for full bug bounty payment
- Tag 2.0.0-pre.1
- Push NPM 2.0.0-pre.1
- Merge to master

**Comment by fulldecent** [comment]
Can you please pull the branch from the @MoMannn fork into the 0xcert repo? 

I want to make some PRs against it but I found a bug in GitHub where I can't make a PR to your fork.

File: src/contracts/utils/address-utils.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.5.1;
```
Comment (fulldecent) [inline src/contracts/utils/address-utils.sol]: Give credit to https://github.com/OpenZeppelin/openzeppelin-solidity/blob/f7e53d90fa638553ffc93e93fe1b12fc081bb774/contracts/utils/Address.sol

**Comment by fulldecent** [comment]
I completed my review. This is a code-level review and I have not looked at test coverage or test accuracy. I do want to look at tests because they have changed.

---

This is good to release as 2.0.0-pre.1. I have several comments and fixes to propose. Maybe some of them can deliver by 2.0.0 final or some can come 2.0.1. None of changes I'm considering relate to ERC-721 mechanics. But some of them do touch actual Solidity code.

Due to a bug in GitHub I can't put in all these PRs against @MoMannn's branch. Once the code is in here then I can start to make PRs.

**Comment by MoMannn** [comment]
@fulldecent It is merged in: https://github.com/0xcert/ethereum-erc721/tree/v2 so fire away :)

**Pull request merged**